app/src/main/kotlin/com/wire/android/util/FileManager.kt

```diff
@@ -28,15 +30,16 @@ class FileManager @Inject constructor(@ApplicationContext private val context: C
         shareAssetFileWithExternalApp(assetDataPath, context, assetExtension, onError)
     }
 
-    fun copyToTempPath(uri: Uri, tempCachePath: Path): Long {
-        val file = tempCachePath.toFile()
-        var size: Long
-        file.setWritable(true)
-        context.contentResolver.openInputStream(uri).use { inputStream ->
-            file.outputStream().use {
-                size = inputStream?.copyTo(it) ?: -1L
+    suspend fun copyToTempPath(uri: Uri, tempCachePath: Path, dispatcher: DispatcherProvider = DefaultDispatcherProvider()): Long =
```

Comment:
@gongracr what is this change?

Reply:
That's injecting a custom coroutine context dispatcher, so that we can make sure the IO operation to copy the data to a file path, is executed on the right IO thread. I made the change after I saw some warnings from the disk strict policy logger when testing the backup restore.